File: tests/test_autocorr.py
from bairstow.autocorr import extract_autocorr, initial_autocorr, pbairstow_autocorr
from bairstow.rootfinding import find_rootq


def test_autocorr():
    h = [10.0, 34.0, 75.0, 94.0, 150.0, 94.0, 75.0, 34.0, 10.0]
    vr0s = initial_autocorr(h)
    vrs, niter, found = pbairstow_autocorr(h, vr0s)
    print([niter, found])

    for vr in vrs:
        vr = extract_autocorr(vr)

    print(find_rootq(vr) for vr in vrs)

    assert niter <= 7

File: tests-pyrs/test_autocorr.rs
use std::collections::HashMap;
use std::*;

use bairstow::autocorr::{extract_autocorr, initial_autocorr, pbairstow_autocorr};
use bairstow::rootfinding::{find_rootq, Options};
fn test_autocorr() {
    let h = vec![10.0, 34.0, 75.0, 94.0, 150.0, 94.0, 75.0, 34.0, 10.0];
    let vr0s = initial_autocorr(h);
    let opts = Options();
    let (vrs, niter, found) = pbairstow_autocorr(h, vr0s, opts);
    println!("{:?} ", vec![niter, found]);
    for vr in vrs {
        vr = extract_autocorr(vr);
        println!("{:?} ", find_rootq(vr));
    }
    assert!(niter <= 13);
}
